x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/In.java

```diff
@@ -345,6 +383,40 @@ static boolean process(BitSet nulls, BitSet mvs, long lhs, long[] rhs) {
         return false;
     }
 
+    /**
+     * Processor for mixed millisecond and nanosecond dates, where the "value" (aka lhs) is in nanoseconds
+     * and the "list" (aka rhs) is in milliseconds
+     */
+    static boolean processNanosMillis(BitSet nulls, BitSet mvs, long lhs, long[] rhs) {
+        for (int i = 0; i < rhs.length; i++) {
+            if ((nulls != null && nulls.get(i)) || (mvs != null && mvs.get(i))) {
+                continue;
+            }
+            Boolean compResult = DateUtils.compareNanosToMillis(lhs, rhs[i]) == 0;
+            if (compResult == Boolean.TRUE) {
+                return true;
+            }
```

Comment:
I think we don't need the boxing. `Comparisons.eq()` used in other methods can return `null` (though not sure if at the respective call sites this can actually occur).

```suggestion
            if (DateUtils.compareNanosToMillis(lhs, rhs[i]) == 0) {
                return true;
            }
```